pkgs/tools/security/semgrep/semgrep-core.nix

```diff
@@ -1,21 +1,48 @@
-{ lib, stdenvNoCC, callPackage }:
+{ lib, stdenvNoCC, fetchPypi, unzip }:
 
 let
-  common = callPackage ./common.nix { };
+  common = import ./common.nix { inherit lib; };
 in
 stdenvNoCC.mkDerivation rec {
   pname = "semgrep-core";
   inherit (common) version;
-  inherit (common.core) src;
+  # fetch pre-built semgrep-core since the ocaml build is complex and relies on
+  # the opam package manager at some point
+  # pulling it out of the python wheel as r2c no longer release a built binary
+  # on github releases
+  src =
+    let
+      inherit (stdenvNoCC.hostPlatform) system;
+      data = common.core.${system} or (throw "Unsupported system: ${system}");
+    in
+    fetchPypi rec {
+      pname = "semgrep";
+      inherit version;
+      format = "wheel";
+      dist = python;
+      python = "cp37.cp38.cp39.py37.py38.py39";
+      inherit (data) platform hash;
+    };
+
+  nativeBuildInputs = [
+    unzip
+  ];
+
+  unpackPhase = ''
+    runHook preUnpack
+    unzip $src
+    runHook postUnpack
+  '';
```

Comment:
Isn't this the default when unzip is added to nativeBuildInputs?

Reply:
Not sure. I thought it got upset because the extension is .whl

I can try again without

Reply:
Tried it and yeah the unzip unpacker checks for a `.zip` extension.

I guess we could modify it to accept `.whl` and maybe `.ZIP` while we're at it

https://github.com/NixOS/nixpkgs/blob/master/pkgs/tools/archivers/unzip/setup-hook.sh#L3

Reply:
For now I have added a comment